test/e2e/node_reuse_test.go

```diff
@@ -473,6 +473,13 @@ func node_reuse() {
 		e2eConfig.GetIntervals(specName, "wait-machine-running")...,
 	).Should(Equal(numberOfAllBmh))
 
+	By("Update Metal3MachineTemplate nodeReuse field to 'False'")
+	m3machineTemplateName = fmt.Sprintf("%s-controlplane", clusterName)
+	updateNodeReuse(false, m3machineTemplateName, targetClusterClient)
+	m3machineTemplateName = fmt.Sprintf("%s-controlplane", clusterName)
+	updateNodeReuse(false, m3machineTemplateName, targetClusterClient)
```

Comment:
even though this should not be needed anymore, this is setting node reuse to false in the same m3mt

Reply:
we discussed with mohamed that this PR would go in like this then when he refactors the node_reuse this will be removed in that PR.

Reply:
That refactoring will covered in #356

Reply:
@Rozzii and what happens if you do not set this to False in this PR? it should pass anyways because current implementation of e2e node reuse tests is not covering image upgrade part and following the old test workflow(where there is no new m3mt creation is introduced)

Reply:
It fails if it is not set to false, let's discuss this offline because I want to reflect on https://github.com/metal3-io/metal3-dev-env/pull/855 too but I haven't tested it yet. But anyways if it is not set to false it fails or at least it has failed until now, I will run a local test again just to make sure.

Reply:
It is although true that there is a duplication by accident I will remove that.

Reply:
That might means the issue raise only when CP is true ?

Reply:
Could be, we will now more after the local test has finished, I don't want to stress the CI at the moment.

Reply:
> That might mean the issue raise even if only the CP is true ?

I bet no, this should be removed completely and tests pass if we are following the exact same test flow as in m3-dev-env or previous workflow of it which does not include image upgrade (current node reuse e2e tests follow latter).

Reply:
It looks like the issue fully disappeared, I have removed this and as @furkatgofurov7  mentioned it has passed, but then i don't get it why wasn't it passing on v1alpha5.

Reply:
So just to summarize the findings:

So this part of code that is highlighted above was locally (and partially in CI when it has worked) tested with both setting the CP and the worker nodes "node reuse" value to false and that was the only thing that made the test pass.

In the current PR i have just accidentally duplicated the CP node updating logic and deleted the worker node related part so this wasn't intentional!

Now in v1beta1 if  the "node reuse" value is not to false at the end of the nodeReuse test the test still passes, we have tested it on at least 4 machine both centos and ubuntu and in v1alpha5 it didn't work and now it does.